src/common/session/session-entity.js

```diff
@@ -0,0 +1,225 @@
+import { generateRandomHexString } from '../ids/unique-id'
+import { warn } from '../util/console'
+import { stringify } from '../util/stringify'
+import { documentAddEventListener } from '../event-listener/event-listener-opts'
+import { ee } from '../event-emitter/contextual-ee'
+import { subscribeToVisibilityChange } from '../window/page-visibility'
+import { Timer } from '../timer/timer'
+import LocalStorage from '../storage/local-storage.js'
+import FPC from '../storage/first-party-cookies'
+import { getConfiguration } from '../config/config'
+
+const PREFIX = 'NRBA'
+
+export class SessionEntity {
```

Comment:
This class should probably be broken up. Create a `SessionManager` that is responsible for creating and managing a session, the timers, and where a session is stored. A `SessionEntity` is just a `key=value` pair tied to a specific `SessionManager`. The concern here is this class is trying to do too much and, for multiple `key=value` pairs, we will have multiple timers running that are not in sync.

Reply:
We should discuss because I would argue that at some point we probably should be able to have multiple session entities with different timer rules

Reply:
currently, the session-entity is tied to one key, but can have many values (the main value, but other custom values too).  You could have multiple session-entities, with distinct values and timers and rules.  we'll need to discuss the shortcomings of this

Reply:
Ok so maybe I was thinking of the key in the wrong way. I think we need some docs around what constitutes a session key or entity.

Reply:
"Session" is a session entity instance, with a key value and potentially user-defined data. 

Later on something else in the agent like "CheckoutSession" or "ClickSession" or whatever could be added to the agent with completely different rules from a "Session", but with the concept that it expires in a certain amount of time. 

A "session entity" in general (beyond the concept of a "session"), is just a key and a value -- stored somewhere (local storage by default)... but it has some extra "housekeeping" data saved with it like timestamps and can have custom user data added to it through the newrelic api